coderd/database/queries/workspaces.sql

```diff
@@ -145,6 +145,135 @@ OFFSET
     @offset_
 ;
 
+-- this duplicates the filtering in GetWorkspaces
+-- name: GetWorkspaceCount :one
```

Comment:
I left an ad-hoc comment on the commit, but I think this is quite bug-prone.

It's not extremely obvious that you *have* to change this when introducing a new filter type, which could lead to a mismatch with the workspaces and count.

Reply:
Yeah, Colin thought this was the best thing for now but we have an issue to clean it up https://github.com/coder/coder/issues/4604